libsupport/include/katana/EntityTypeManager.h

```diff
@@ -39,6 +40,20 @@ using AtomicTypeNameToEntityTypeIDMap =
 using EntityTypeIDToAtomicTypeNameMap =
     std::unordered_map<EntityTypeID, std::string>;
 using TypeNameSet = std::set<std::string>;
+// Hash function when TypeNameSet is a key
+struct TypeNameSetHash {
```

Comment:
This can be expensive if we are using the hash function on very large maps/sets, e.g. O(V) or O(E). I'd like to encourage an API that works with type DI sets (`std::bitset`) instead of type name sets (`set<string>`)

Reply:
Type name sets are small: order of number of properties per node (10s) or edge (1 in Cypher).

I need a set of strings interface because that is what I get from the Cypher engine.  If you or @roshandathathri tell them to pass me bitsets then I can recode my public API to only accept bitsets.  But I think that might have problems with bitsets going stale as label sets change.  The last time we had this discussion, Roshan called the string set the source of truth.

Reply:
+1 for `std::set<std::string>` as the source of truth/canonical form.

Reply:
Yes, the source of truth is a set of strings.

Reply:
Is this needed for `std::set`? I thought not. It's needed for `std::unordered_set` or `std::unordered_hash`. Can we avoid using those?